tests/phpunit/integration/tests/Story_Post_Type.php

```diff
@@ -98,6 +101,30 @@ public function test_get_post_type_icon() {
 		$this->assertContains( 'data:image/svg+xml;base64', $valid );
 	}
 
+	/**
+	 * @covers ::register_post_type
+	 */
+	public function test_register_post_type() {
+		$story_post_type = $this->get_story_object();
+		$post_type       = $story_post_type->register_post_type();
+		$this->assertTrue( $post_type->has_archive );
+		update_option( \Google\Web_Stories\Settings::SETTING_NAME_ARCHIVE, 'disabled' );
+		$story_post_type->unregister_post_type();
+		$post_type = $story_post_type->register_post_type();
+		$this->assertFalse( $post_type->has_archive );
+		delete_option( \Google\Web_Stories\Settings::SETTING_NAME_ARCHIVE );
```

Comment:
There should ideally be no code following assertions. If the assertion fails, the `update_option()` and `delete_option()` calls are never made, which means they'll leak to other tests.

I'd suggest splitting this test into two and then either move the `delete_option()` call to `tearDown()` or refactor the test like this:

```php
$story_post_type->unregister_post_type();
$post_type = $story_post_type->register_post_type();
$has_archive = $post_type->has_archive;
delete_option( \Google\Web_Stories\Settings::SETTING_NAME_ARCHIVE );
$this->assertFalse( $has_archive );
```